### PR TITLE
147 add new module canpb

### LINKED ISSUE #147: Add new module canpb
This module is used for communication inside the robot, like for uartpb, but on CAN bus.

### LINKED ISSUE #150: Create new board cogip-board
Create new board for 2024 robot.

### PR DESCRIPTION
**Pull request by gdoffe**
Add new cogip board and its CAN FD support.

Fix #147

### CODE REVIEW

File: sys/canpb/include/canpb/CanProtobuf.hpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        message_handler_t handler                 ///< [in] message handler
+        );
+
+private:
+    conn_can_raw_t conn_can_raw_;                 ///< Raw CAN connection
+    uint8_t can_interface_number_;                ///< CAN interface number
+    kernel_pid_t reader_pid_;                     ///< reader thread PID
+    ringbuffer_t rx_buf_;                         ///< ring buffer for CAN incoming bytes
+    uint32_t msg_length_ = 0;                     ///< message length
+    mutex_t mutex_ = MUTEX_INIT;                  ///< mutex protecting serial port access
```
Comment (ecourtois) [inline sys/canpb/include/canpb/CanProtobuf.hpp:102]: Remaining `serial` in comment.

File: sys/canpb/include/canpb/CanProtobuf.hpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+/// @brief       Exchange Protobuf messages over CAN module.
+/// @details     This module provides a class that controls an CAN to send and
+///              receive Protobuf messages using EmbbededProto.
+///              Each library/application using canpb must define a unique id (32-bit integer
+///              of type cogip::canpb::uuid_t) and registrer a callback function to send
+///              or receive their own messages.
+///              See `examples/canb`.
+/// @{
+/// @file
+/// @author      Eric Courtois <eric.courtois@gmail.com>
```
Comment (ecourtois) [inline sys/canpb/include/canpb/CanProtobuf.hpp:16]: Not me, not alone ;-)
Same for other headers.

**Review by ecourtois — APPROVED** [review]
Just remarks on comments/docs.
Approved anyway!

File: sys/canpb/include/canpb/CanProtobuf.hpp (hunk 2 above)
Comment (gdoffe) [reply sys/canpb/include/canpb/CanProtobuf.hpp:17]: Done

File: sys/canpb/include/canpb/CanProtobuf.hpp (hunk 1 above)
Comment (gdoffe) [reply sys/canpb/include/canpb/CanProtobuf.hpp]: Done

**Pull request merged**